tests/js/dataMock/analytics.js

```diff
@@ -0,0 +1,311 @@
+ *
+ * @since n.e.x.t
+ *
+ * @param {string|Object} metric Metric name or object.
+ * @return {string} Metric key.
+ */
+function getMetricKey( metric ) {
+	return metric?.expression || metric.toString();
+}
+
+/**
+ * Gets metric type.
+ *
+ * @since n.e.x.t
+ *
+ * @param {string|Object} metric Metric name or object.
+ * @return {string} Type of the metric.
+ */
+function getMetricType( metric ) {
+	return ANALYTICS_METRIC_TYPES[ getMetricKey( metric ) ];
+}
+
+/**
+ * Generates and returns metric values.
+ *
+ * @since n.e.x.t
+ *
+ * @param {string} type Metric type.
+ * @param {number} max  Maximum number of values to generate.
+ * @return {Array.<Object>} Array of metric values.
+ */
+function generateMetricValues( type, max ) {
+	const metrics = [];
+
+	for ( let i = 0; i < max; i++ ) {
+		const values = [];
+
+		switch ( type ) {
+			case 'INTEGER':
+				values.push( faker.random.number( { min: 0, max: 100 } ).toString() );
+				break;
+			case 'PERCENT':
+				values.push( faker.random.float( { min: 0, max: 100 } ).toString() );
+				break;
+			case 'TIME':
+				values.push( faker.random.number( { min: 0, max: 3600 } ).toString() ); // 1 hour max.
+				break;
+			case 'CURRENCY':
+				values.push( faker.random.float( { min: 0, max: 10000 } ).toString() ); // $10k max.
+				break;
+		}
+
+		metrics.push( { values } );
+	}
+
+	return metrics;
+}
+
+/**
+ * Sorts report rows and returns it.
+ *
+ * @since n.e.x.t
+ *
+ * @param {Array.<Object>}        rows    Array of rows to sort.
+ * @param {Array.<Object>}        metrics Array of report metrics.
+ * @param {Object|Array.<Object>} orderby Sorting options.
+ * @return {Array.<Object>} Sorted rows.
+ */
+function sortRows( rows, metrics, orderby ) {
+	let sorted = rows;
+
+	const orders = Array.isArray( orderby ) ? orderby : [ orderby ];
+	for ( const order of orders ) {
+		const direction = order?.sortOrder === 'DESCENDING' ? -1 : 1;
+		const index = metrics.findIndex( ( metric ) => getMetricKey( metric ) === order?.fieldName );
+		if ( index < 0 ) {
+			continue;
+		}
+
+		sorted = sorted.sort( ( a, b ) => {
+			let valA = parseFloat( a.metrics[ index ]?.values?.[ 0 ] );
+			if ( Number.isNaN( valA ) ) {
+				valA = 0;
+			}
+
+			let valB = parseFloat( b.metrics[ index ]?.values?.[ 0 ] );
+			if ( Number.isNaN( valB ) ) {
+				valB = 0;
+			}
+
+			return ( valA - valB ) * direction;
+		} );
+	}
+
+	return sorted;
+}
+
+/**
+ * Generates mock data for Analytics reports.
+ *
+ * @since n.e.x.t
+ *
+ * @param {Object} args Report options.
+ * @return {Array.<Object>} An array with generated report.
+ */
+export function getAnalyticsMockResponse( args ) {
+	// We set seed for every data mock to make sure that the same arguments get the same report data.
+	// It means that everyone will have the same report data and will see the same widgets in the storybook.
+	// This approach gives us additional flexebility to control randomness on per widget basis.
+	faker.seed( getObjectHash( args ) );
```

Comment:
> I think this would result in more realistic results where different sites would have very different reports, but reports for the same URL should be largely the same.

What to do when the `args.url` argument is not provided (not used by a report)?

Reply:
> What to do when the `args.url` argument is not provided (not used by a report)?

In this case we could use a default value instead (e.g. `http://example.com`) since reports use the context of the reference URL if no `url` arg is provided.

Reply:
I prefer to rely on the whole object hashes instead of using URLs only. Anyway, updated it to use URLs instead of objects.